File: 1047-remove-all-adjacent-duplicates-in-string/python/Solution.py
class Solution(object):
    def removeDuplicates(self, S):
        """
        :type S: str
        :rtype: str
        """
        stack = list()
        for ch in S:
            if stack and stack[-1] == ch:
                stack.pop()
            else:
                stack.append(ch)
        return "".join(stack)

File: 1047-remove-all-adjacent-duplicates-in-string/javascript/Solution.js
var removeDuplicates = function (S) {
  const stack = [];
  for (const ch of S) {
    if (stack.length && stack[stack.length - 1] === ch) {
      stack.pop();
    } else {
      stack.push(ch);
    }
  }
  return stack.join('');
};